report/renderer/i18n.js

```diff
@@ -22,25 +22,58 @@ export class I18n {
     // When testing, use a locale with more exciting numeric formatting.
     if (locale === 'en-XA') locale = 'de';
 
-    this._numberDateLocale = locale;
-    this._numberFormatter = new Intl.NumberFormat(locale);
-    this._percentFormatter = new Intl.NumberFormat(locale, {style: 'percent'});
+    this._locale = locale;
     this._strings = strings;
   }
 
   get strings() {
     return this._strings;
   }
 
+  /**
+   * @param {number} number
+   * @param {number} granularity
+   * @param {Intl.NumberFormatOptions} opts
+   * @return {string}
+   */
+  _formatNumberWithGranularity(number, granularity, opts = {}) {
+    opts = {...opts};
```

Comment:
any particular reason to be this defensive? The DevTools repl, at least, treats `new Intl.NumberFormat(this._locale, opts)` as side-effect-free

Reply:
yeah i had my cs prof yelling in my head "don't mutate input parameters!" so i copied it